python/python-core-impl/src/com/jetbrains/python/extensions/QualifiedNameExt.kt

```diff
@@ -142,12 +142,57 @@ fun QualifiedName.getElementAndResolvableName(context: QNameResolveContext, stop
   if (lastElement != null && element is PyClass) {
     // Drill in class
 
-    //TODO: Support nested classes
+    // First try to find nested class
+    val nestedClass = element.findNestedClass(lastElement, true)
+    if (nestedClass != null) {
+      val newName = currentName.append(lastElement)
+      // If this is the last component, return the nested class
+      if (newName.componentCount == this.componentCount) {
+        return NameAndElement(newName, nestedClass)
+      }
+      // If there are more components, try to resolve them starting from the nested class
+      val remainingName = subQualifiedName(newName.componentCount, componentCount)
+
+      // Try to find a method in the nested class if this is the last component
+      if (remainingName.componentCount == 1) {
+        val method = nestedClass.findMethodByName(remainingName.lastComponent!!, true, context.evalContext)
+        if (method != null) {
+          return NameAndElement(this, method)
+        }
+      }
+
+      // Try to find next level nested class
+      val nextComponent = remainingName.firstComponent
+      if (nextComponent != null) {
+        val nextNestedClass = nestedClass.findNestedClass(nextComponent, true)
+        if (nextNestedClass != null) {
+          // If we found a nested class, continue resolving from it
+          val remainingAfterNext = remainingName.subQualifiedName(1, remainingName.componentCount)
+          if (remainingAfterNext.componentCount == 0) {
+            return NameAndElement(this, nextNestedClass)
```

Comment:
This `return` is not covered by the tests

Reply:
testNestedInnerClass covers this now.